base/utf8.jl

```diff
@@ -112,8 +103,11 @@ function getindex(s::UTF8String, r::UnitRange{Int})
     isempty(r) && return empty_utf8
     i, j = first(r), last(r)
     d = s.data
+    if i < 1 || i > length(s.data)
```

Comment:
Yeah, would probably be clearer and faster to write that as sizeof(s), since this is a boundscheck, not a UTF8 validation.